crates/rust-analyzer/src/caps.rs

```diff
@@ -40,7 +40,19 @@ pub fn server_capabilities() -> ServerCapabilities {
         document_highlight_provider: Some(true),
         document_symbol_provider: Some(true),
         workspace_symbol_provider: Some(true),
-        code_action_provider: Some(CodeActionProviderCapability::Simple(true)),
+        code_action_provider: Some(CodeActionProviderCapability::Options(CodeActionOptions {
+            code_action_kinds: Some(vec![
+                "".to_string(),
```

Comment:
this probably needs comment (and could be `String::new()`)

Reply:
I will make a PR to lsp-types to put in `Empty`